lib/ansible/modules/cloud/amazon/inspector_target.py

```diff
@@ -0,0 +1,238 @@
+tags:
+  description: The tags of the resource group that is associated with the
+               assessment target.
+  returned: success
+  type: list
+  sample: [{"key": "role", "value": "scan_target"},
+           {"key": "env", "value": "dev"}]
+updated_at:
+  description: The time at which the assessment target was last updated.
+  returned: success
+  type: string
+  sample: "2018-01-29T13:48:51.958000+00:00"
+'''
+
+try:
+    from botocore.exceptions import ClientError, ValidationError
+except ImportError:
+    pass  # Handled AnsibleAWSModule
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import AWSRetry
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    camel_dict_to_snake_dict,
+    get_aws_connection_info,
+)
+
+
+@AWSRetry.backoff()
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        state=dict(choices=['absent', 'present'], default='present'),
+        tags=dict(type='dict'),
+    )
+
+    required_if = [['state', 'present', ['tags']]]
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_if=required_if,
+    )
+
+    name = module.params.get('name')
+    state = module.params.get('state').lower()
+    tags = module.params.get('tags')
+    if tags:
+        tags = [{'key': key, 'value': value} for key, value in tags.items()]
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(
+        module,
+        boto3=True,
+    )
+
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    try:
+        client = boto3_conn(
+            module, conn_type='client',
+            resource='inspector',
+            region=region,
+            endpoint=ec2_url,
+            **aws_connect_kwargs
+        )
+    except (ClientError, ValidationError) as e:
+        module.fail_json_aws(e, msg="trying to connect to AWS")
+
+    try:
+        existing_target_arn = client.list_assessment_targets(
+            filter={'assessmentTargetNamePattern': name},
+        ).get('assessmentTargetArns')[0]
+
+        existing_target = camel_dict_to_snake_dict(
+            client.describe_assessment_targets(
+                assessmentTargetArns=[existing_target_arn],
+            ).get('assessmentTargets')[0]
+        )
+
+        existing_resource_group_arn = existing_target.get('resource_group_arn')
+        existing_resource_group_tags = client.describe_resource_groups(
+            resourceGroupArns=[existing_resource_group_arn],
+        ).get('resourceGroups')[0].get('tags')
+
+        target_exists = True
+    except (ClientError, ValidationError) as e:
+        module.fail_json_aws(e, msg="trying to retrieve targets")
+    except IndexError:
+        target_exists = False
+
+    if state == 'present' and target_exists:
+        if sorted(tags) == sorted(existing_resource_group_tags):
```

Comment:
This will fail on python 3 with a TypeError because dicts are unorderable types. You could import compare_aws_tags from ansible.module_utils.ec2 and do something like:
```
        #if sorted(tags) == sorted(existing_resource_group_tags):
        tags_to_add, tags_to_remove = compare_aws_tags(boto3_tag_list_to_ansible_dict(tags),
                                                       boto3_tag_list_to_ansible_dict(existing_resource_group_tags))
        if not (tags_to_add or tags_to_remove):
```